### PR TITLE
Semantic fork versions and signature domains

### PR DESCRIPTION
**Pull request by JustinDrake**
Address #706

### CODE REVIEW

**Comment by hwwhww** [comment]
It requires BLS signature spec API updates and I think we agreed not changing that before the standardized spec?

**Comment by JustinDrake** [comment]
@hwwhww No changes to BLS spec required now 👍

File: specs/core/0_beacon-chain.md [hunk 1]
```diff
@@ -1107,7 +1107,7 @@ def get_total_balance(state: BeaconState, validators: List[ValidatorIndex]) -> G
 
 ```python
 def get_fork_version(fork: Fork,
-                     epoch: Epoch) -> int:
+                     epoch: Epoch) -> bytes4:
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]: Ah sorry, one sec, we haven't defined `Bytes4` custom type yet, so either changed this line to
```suggestion
                     epoch: Epoch) -> bytes:
```
Or defined `Bytes4` and then here we can do:
```suggestion
                     epoch: Epoch) -> Bytes4:
```

What do you think about it? :)

File: specs/core/0_beacon-chain.md [hunk 2]
```diff
@@ -1122,12 +1122,11 @@ def get_fork_version(fork: Fork,
 ```python
 def get_domain(fork: Fork,
                epoch: Epoch,
-               domain_type: int) -> int:
+               domain_type: bytes4) -> int:
```
Comment (hwwhww) [inline specs/core/0_beacon-chain.md]: ditto, either `domain_type: bytes` or `domain_type: Bytes4`.

File: specs/core/0_beacon-chain.md (hunk 1 above)
Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: `bytesN` (hence `bytes4`) is defined [here](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/simple-serialize.md#aliases). Defining `Bytes4` also is fine I guess.

Comment (djrtwo) [reply specs/core/0_beacon-chain.md]: `bytes4` isn't valid python outside of the string context unless we define a new type in custom types. I think I'd rather just leave this as `bytes`

Comment (JustinDrake) [reply specs/core/0_beacon-chain.md]: Done! Also makes the diff super small.

**Pull request merged**